client/internal/vanus/store/block_store.go

```diff
@@ -94,16 +94,15 @@ func (s *BlockStore) Append(ctx context.Context, block uint64, event *ce.Event)
 
 	res, err := client.(segpb.SegmentServerClient).AppendToBlock(_ctx, req)
 	if err != nil {
-		vlog.Warning(ctx, "failed to AppendToBlock", map[string]interface{}{
-			vlog.KeyError: err,
-			"block_id":    block,
-		})
-		sts := status.Convert(err)
-		// TODO: temporary scheme, wait for error code reconstruction
-		if strings.Contains(sts.Message(), "SEGMENT_FULL") {
-			return -1, errors.ErrNoSpace
+		if errStatus, ok := status.FromError(err); ok {
```

Comment:
Could add a function in `pkg/errors` like
```go
func Is(err error, code errpb.ErrorCode) bool {
	if errStatus, ok := status.FromError(err); ok {
		if errType, ok := errpb.Convert(errStatus.Message()); ok {
			return errType.Code == code
		}
	}
	return false
}
```
and replace here as `errors.Is(err, errpb.ErrorCode_SEGMENT_FULL)`?